datafusion/physical-plan/src/joins/hash_join.rs

```diff
@@ -688,11 +770,21 @@ impl ExecutionPlan for HashJoinExec {
             );
         }
 
+        let shared_state = context.session_config().get_extension::<SharedJoinState>();
+
         let join_metrics = BuildProbeJoinMetrics::new(partition, &self.metrics);
         let left_fut = match self.mode {
             PartitionMode::CollectLeft => self.left_fut.once(|| {
                 let reservation =
                     MemoryConsumer::new("HashJoinInput").register(context.memory_pool());
+
+                let probe_threads = shared_state
```

Comment:
If shared state is available we use the `num_task_partitions` instead of output partitioning to determine the number of local probe threads